### PR TITLE
Initial re-write for contributor issue triage guide

### PR DESCRIPTION
**Pull request by jmhbnz**
This pull request proposes a significant re-write of our existing etcd issue triage guidelines.

The purpose of the re-write is to make the guidelines into a more step by step process with clearer descriptions of the actual actions expected to be taken. 

Additionally these new guidelines align closer to the style of [kubernetes issue triage guidelines](https://www.kubernetes.dev/docs/guide/issue-triage) and now take into account our new etcd [community membership model](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/community-membership.md).

The expected outcome is that these new guidelines will be easier for contributors to follow and will hopefully empower our new @etcd-io/members to be active with triage, which should reduce burden on maintainers and help support a healthy project.

Relates to https://github.com/etcd-io/etcd/issues/15773

---

**Note for reviewers**: There are a couple of areas namely area labels and prioritization that I am still working on and will be proposing additional changes for in future, I just wanted to get this initial re-write out of the way first so the pr doesn't become too much to review at once.

### CODE REVIEW

File: Documentation/contributor-guide/triage_issues.md
Comment (jmhbnz) [inline Documentation/contributor-guide/triage_issues.md:88]: This `area/etcdutl` label doesn't currently exist but I think should. Can a maintainer please create it if they agree?

File: Documentation/contributor-guide/triage_issues.md
Comment (fuweid) [inline Documentation/contributor-guide/triage_issues.md:38]: what about adding label named by area/external? The dependency bug can cause unexpected behavior in etcd. Marked it external might help to track.

File: Documentation/contributor-guide/triage_issues.md
Comment (fuweid) [inline Documentation/contributor-guide/triage_issues.md:93]: It seems like it's been migrated to robustness testing.

Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md]: Good spotting - removed :)

File: Documentation/contributor-guide/triage_issues.md
Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md:38]: Makes sense to me - gives us a way to recognise over time if there are common types of issues that are being raised against etcd incorrectly so we can do something more about it. Added `area/external`.

Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md:38]: Will need a maintainer to create the new label if they agree.

File: Documentation/contributor-guide/triage_issues.md
Comment (fuweid) [inline Documentation/contributor-guide/triage_issues.md:28]: help-wanted can be considered as well ：）

**Review by fuweid — APPROVED** [review]
LGTM

File: Documentation/contributor-guide/triage_issues.md
Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md:28]: Done - Added new filter and section covering `help wanted` and `good first issue`, based on simplified version of kubernetes docs here https://www.kubernetes.dev/docs/guide/help-wanted.

File: Documentation/contributor-guide/triage_issues.md
Comment (serathius) [reply Documentation/contributor-guide/triage_issues.md:90]: SG

**Review by serathius — APPROVED** [review]
Looks like a good base that we can iterate on.

File: Documentation/contributor-guide/triage_issues.md
Comment (chaochn47) [inline Documentation/contributor-guide/triage_issues.md:28]: Even if the issue is not active but if it's not resolved. We can use `stage/tracked` to avoid stale bot close it accidentally. https://github.com/etcd-io/etcd/pull/14108

Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md:28]: Good suggestion!  Added a section under **Step 7 - Follow up** to cover tracking important issues.

File: Documentation/contributor-guide/triage_issues.md
Comment (lavacat) [inline Documentation/contributor-guide/triage_issues.md:21]: idea: have a #triage-wanted label that maintainers can put to delegate triage to members. I assume maintainers skim through new incoming issues. This label will give signal boost. Ideally all issues should be triaged but I don't think that's happening.

Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md:21]: Hmm I see what you mean, though my thinking is that these guidelines are to help all us new members triage in a useful way with **full autonomy**, i.e. without needing maintainers to do anything at all to prod us.  I think we are already starting to do that.

Maybe we could re-evaluate in a month or two after these merge and if there are issues still going without triage for too long then we could add a way for maintainers to prod / suggest we triage?

Some other projects namely kubernetes also by default add labels for needing triage to all new issues, but I think that is less useful on etcd where we don't have any bot automation and there is less volume of issues. 

Curious what others think,  if more folks want to add it straight away I will update and add a section for it.

File: Documentation/contributor-guide/triage_issues.md
Comment (ahrtr) [reply Documentation/contributor-guide/triage_issues.md:90]: Agreed & thx & created

File: Documentation/contributor-guide/triage_issues.md
Comment (ahrtr) [reply Documentation/contributor-guide/triage_issues.md:39]: It seems not necessary, because eventually one of etcd area/component (e.g. etcdserver, etcdctl, etc) will run into issue. But I have no objection to add one more label `area/external`. Leave it to @serathius to decide.

File: Documentation/contributor-guide/triage_issues.md
Comment (ahrtr) [inline Documentation/contributor-guide/triage_issues.md:31]: I suggest to add one more invalid case which is a false alarm. e.g. It might caused by some misunderstanding (e.g. the reporter might think it's a bug, but actually it's expected behavior) or misconfiguration.

Comment (jmhbnz) [reply Documentation/contributor-guide/triage_issues.md:31]: Just to double check, we do have this section under bug triage:

> ### Bug reports
>
> If an issue has been raised as a bug it should already have the `type/bug` label, however if this is missing for an issue you determine to be a bug please add the label manually.
>
> The next step is to validate if the issue is indeed a bug. If not, add a comment with findings and close trivial issue. For non-trivial issue, wait to hear back from issue reporter and see if there is any objection. If issue reporter does not reply in 30 days, close the issue.

Do you think this can cover the invalid bug situation enough?

**Comment by ahrtr** [comment]
Thanks @jmhbnz . Overall looks good to me with a couple of minor comments.

One more general comment, it seems that step 3 ~ 6 belong to one step with a couple of sub-sections:

**Apply appropriate labels**
- Identify the type 
- identify the area
- Identify the priority
- Support new contributors

File: Documentation/contributor-guide/triage_issues.md
Comment (ahrtr) [reply Documentation/contributor-guide/triage_issues.md:31]: ack

**Comment by ahrtr** [comment]
@jmhbnz Please let me know if you want to discuss or address the comment separately. If YES, I can merge the PR now.

**Comment by jmhbnz** [comment]
> @jmhbnz Please let me know if you want to discuss or address the comment separately. If YES, I can merge the PR now.

Hey @ahrtr - I'm going to be following up with some later changes to tackle things like the prioritization section so I don't mind merging this first iteration as is. With the help of all the feedback above I think we have it in pretty decent shape.  I'll have a think about your feedback on number of steps and have a tinker with that layout when I add the priority instructions :)

**Review by ahrtr — APPROVED** [review]

**Pull request merged**
